src/python-common/ceph/deployment/hostspec.py

```diff
@@ -42,34 +43,55 @@ def __init__(self,
         #: human readable status
         self.status = status or ''  # type: str
 
+        self.location = location
+
     def to_json(self) -> dict:
         return {
             'hostname': self.hostname,
             'addr': self.addr,
             'labels': list(OrderedDict.fromkeys((self.labels))),
             'status': self.status,
+            'location': self.location,
```

Comment:
how about avoiding returning `None` here` this makes the yaml a bit more readable